src/urg_stamped.cpp

```diff
@@ -334,6 +334,51 @@ void UrgStampedNode::cbII(
   {
     scip2::logger::debug() << "sensor status: " << stat->second << std::endl;
   }
+  const auto mesm = params.find("MESM");
+  if (mesm != params.end())
+  {
+    scip2::logger::debug() << "measurement status: " << mesm->second << std::endl;
+  }
+
+  if (delay_estim_state_ == DelayEstimState::STATE_CHECKING)
+  {
+    if (mesm == params.end())
+    {
+      scip2::logger::error() << "II doesn't have measurement state" << std::endl;
+      errorCountIncrement();
+    }
+    else
+    {
+      // MESM (measurement state) value depends on the sensor model.
+      //   UTM-30LX-EW: "Idle"
+      //   UST-**LX: "Measuring"
+      std::string state(mesm->second);
+      const auto tolower = [](unsigned char c)
+      {
+        return std::tolower(c);
+      };
+      std::transform(state.begin(), state.end(), state.begin(), tolower);
+      if (state.find("idle") != std::string::npos || state.find("measuring") != std::string::npos)
+      {
+        if (last_measurement_state_ != state && last_measurement_state_ != "")
+        {
+          scip2::logger::info() << "Sensor is idle, entering time synchronization mode" << std::endl;
+        }
+        delay_estim_state_ = DelayEstimState::ESTIMATION_STARTING;
+        retryTM();
+      }
+      else
+      {
+        if (last_measurement_state_ != state)
+        {
+          scip2::logger::info() << "Sensor is not idle (" << state << ")" << std::endl;
+        }
+        delay_estim_state_ = DelayEstimState::STOPPING_SCAN;
```

Comment:
How do we recover from this state? Is it through https://github.com/seqsense/urg_stamped/blob/9a751b9635a2b794144764821261b6b48c566b15/src/urg_stamped.cpp#L447 or https://github.com/seqsense/urg_stamped/blob/9a751b9635a2b794144764821261b6b48c566b15/src/urg_stamped.cpp#L571?

Reply:
`retryTM` is called by an interval timer and `retryTM` will send `QT` command if `delay_estim_state_ == DelayEstimState::STOPPING_SCAN`